sklearn/callback/tests/_utils.py

```diff
@@ -49,20 +79,23 @@ def __init__(self, max_iter=20, computation_intensity=0.001):
         self.max_iter = max_iter
         self.computation_intensity = computation_intensity
 
-    @_fit_context(prefer_skip_nested_validation=False)
-    def fit(self, X=None, y=None):
-        callback_ctx = self.init_callback_context(
-            max_subtasks=self.max_iter
-        ).eval_on_fit_begin(estimator=self)
-
+    def __skl_fit__(
+        self, X_train=None, y_train=None, X_val=None, y_val=None, callback_ctx=None
+    ):
         for i in range(self.max_iter):
-            subcontext = callback_ctx.subcontext(task_id=i)
+            if callback_ctx is not None:
```

Comment:
We need a new method, like `set_task_info`, to set `max_subtasks`, `task_name` and `task_id` here, now that `init_callback_context` has moved.
```py
callback_ctx.set_task_info(...)
```